server/src/pages/shotindex/controller.js

```diff
@@ -213,14 +216,6 @@ exports.toggleFavoriteShot = async function(shot) {
     render();
   } catch (err) {
     console.warn("Error updating expiration:", err);
-
-    // The error could be one from the server or a network level error from
-    // fetch; we display a generic message either way as it's (probably?)
-    // friendlier than percolating a fetch error message up to the user.
-    const errorMessage = document.getElementById("shotIndexAlertErrorFavoriteShot").textContent;
-    if (errorMessage) {
-      window.alert(errorMessage);
-    }
```

Comment:
Code above is carried over from shot page and is redundant on myshots

Reply:
@punamdahiya I'm not sure this is redundant. It's good to be consistent in how error messages are presented to the user, and the shot page shows a window.alert error if changing expiration fails, so I think it makes sense to also show a window.alert with an error message here.

It does seem that there is a ton of duplication between this controller's `toggleFavoriteShot` and the shot index controller's `changeShotExpiration` (which uses xhr, but seems otherwise pretty much identical). If you have the time and inclination, I think it would be great to extract the shared code.

Reply:
  👍 what should be the best place to bring in this common code from controller, a new util class or may be shared/shot

Reply:
Hmm. I think a separate util is the way to go. shared/shot is shared between server and client, so we have to be super careful about what code goes in there.